### DISCUSSION TITLE
Can't format error | onError doesn't get called

### DISCUSSION THREAD

**Discussion by UchihaYuki** [enisdenjo/graphql-ws#569] [Q&A]
**Screenshot**
![image](https://github.com/enisdenjo/graphql-ws/assets/38900426/82bf7285-e27a-4ea9-a682-a96c9087fa14)
![image](https://github.com/enisdenjo/graphql-ws/assets/38900426/2447f9f2-210f-4fb6-b401-2e109ad234db)

**Expected Behaviour**
I hope when an error is thrown, onError can get called.

**Actual Behaviour**
never get called

**Debug Information**
Help us debug the bug?

**Further Information**
Anything else you might find helpful.

**Comment by enisdenjo** [comment]
That is because onError is not handling exceptions in the subscription iterator. [There is an open discussion in graphql-js about what to do.](https://github.com/graphql/graphql-js/issues/4001)

You can read more and handle the thrown error gracefully following the ["ws server usage with custom subscribe method that gracefully handles thrown errors" recipe](https://the-guild.dev/graphql/ws/recipes#ws-server-usage-with-custom-subscribe-method-that-gracefully-handles-thrown-errors).
